src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,49 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int AMOUNT_OF_FIGURES = 5;
+    private static final int MAX_RANDOM_VALUE = 100;
+    private Random random = new Random();
+    private int numOfFigure;
+    private String color;
+    private ColorSupplier colorSupplier;
+
+    public Figure getRandomFigure() {
+        Figure figure;
+        numOfFigure = random.nextInt(AMOUNT_OF_FIGURES);
+        colorSupplier = new ColorSupplier();
+        color = colorSupplier.getRandomColor();
+        switch (numOfFigure) {

```

Comment:
We should create additional variables when we really need them for logic or when introducing additional variable with some specific name will increase readability (like color here, it really makes further code more readable so it's fully ok). In other cases we may use expression itself:
```suggestion
        color = colorSupplier.getRandomColor();
        switch (random.nextInt(AMOUNT_OF_FIGURES)) {
```